openupgrade_scripts/scripts/mail/15.0.1.5/pre-migration.py

```diff
@@ -0,0 +1,66 @@
+from openupgradelib import openupgrade
+
+
+def _copy_columns(env):
+    openupgrade.copy_columns(
+        env.cr,
+        {
+            "mail_activity_type": [
+                ("force_next", None, None),
+                ("res_model_id", None, None),
+            ],
+            "mail_message_res_partner_needaction_rel": [("failure_type", None, None)],
+        },
+    )
+
+
+def _rename_fields(env):
+    openupgrade.rename_fields(
+        env,
+        [
+            (
+                "mail.activity.type",
+                "mail_activity_type",
+                "default_next_type_id",
+                "triggered_next_type_id",
+            ),
+            (
+                "mail.activity.type",
+                "mail_activity_type",
+                "next_type_ids",
+                "suggested_next_type_ids",
+            ),
+            (
+                "mail.activity.type",
+                "mail_activity_type",
+                "default_description",
+                "default_note",
+            ),
+            ("mail.message", "mail_message", "no_auto_thread", "reply_to_force_new"),
+            ("mail.notification", "mail_notification", "mail_id", "mail_mail_id"),
```

Comment:
Slightly puzzled why this seems to work, if it does. mail_notification is the new table name, but this method is called before _rename_tables is.

Reply:
Thanks for the fix! Maybe it's just me, but I prefer to leave 'resolving' comments to the submitter so that I (or other reviewers) can follow up on my own reviews more easily.

Reply:
> 

yes, I think it's reasonable !